### PR TITLE
Drop support for pygments as syntax-highlighter

### LINKED ISSUE #6823: Drop Pygments
Avoid adding a Python dependency just for syntax-highlighting. Default is already to Rouge.

### PR DESCRIPTION
**Pull request by ashmaroli**
Resolves #6823 

- stop testing output from the use of Pygments as syntax highlighter
- maintain API for handling Pygments so that a plugin can easily hook into, if needed in future.
        e.g. The plugin would just have to override `HighlightBlock::render_pygments` to provide the functionality.

Replaces #6983

### CODE REVIEW

File: docs/_docs/templates.md [hunk 1]
```diff
@@ -484,16 +484,23 @@ For more advanced information on using includes, see [Includes](../includes).
...
-`pygments` in your site's configuration file. Pygments supports [over 100
-languages](http://pygments.org/languages/)
+Alternatively, you can use [Pygments](http://pygments.org) to highlight your
+code snippets in Jekyll 3.x and below. To use Pygments, you must have Python
+installed on your system, have the `pygments.rb` gem installed and set
+`highlighter` to `pygments` in your site's configuration file. Pygments
+supports [over 100 languages](http://pygments.org/languages/)
+
+<div class="note info">
+  <p>Using Pygments have been deprecated and will not be officially supported in
```
Comment (oe) [inline docs/_docs/templates.md]: `has been deprecated`

File: docs/_docs/templates.md [hunk 2]
```diff
@@ -484,16 +484,23 @@ For more advanced information on using includes, see [Includes](../includes).
...
+  Jekyll 4, wherein the configuration setting <code>highlighter: pygments</code>
```
Comment (oe) [inline docs/_docs/templates.md]: replace `wherein` with `meaning that`, it sounds better

File: docs/_docs/templates.md [hunk 3]
```diff
@@ -484,16 +484,23 @@ For more advanced information on using includes, see [Includes](../includes).
...
+  will automatically fallback to using <em>Rouge</em> which is written in Ruby
```
Comment (oe) [inline docs/_docs/templates.md]: fall back is two separate words

**Comment by DirtyF** [comment]
@jekyllbot: merge +major

**Comment by ashmaroli** [comment]
Thank you very much for merging this @DirtyF :tada: 
Finally, the test suite will pass successfully at my end..!!

**Pull request merged**